util/dvsim/SimCfg.py

```diff
@@ -48,7 +48,7 @@ def __init__(self, flow_cfg_file, proj_root, args):
         self.max_waves = args.max_waves
         self.cov = args.cov
         self.cov_merge_previous = args.cov_merge_previous
-        self.profile = args.profile
+        self.profile = args.profile or '(cfg uses profile without --profile)'
```

Comment:
The existing code pretty much did the same thing (with `default='none'` arg option). Would you mind explaining why is this necessarily better?

Reply:
Well, the new code is probably more helpful than the old code if something comes unstuck and the config tries to use `{profile}` even though we didn't specify a profile: you now get a "hey, this went wrong" message, rather than it silently doing something silly.

I agree that we could have the same behaviour by changing this line to something like `'some error message' if args.profile == 'none' else args.profile`. That works, but depends on a special string value, rather than using None to signify "was not specified". The latter approach is much more common in Python codebases, and is what argparse does by default, so I think it is probably more sensible.

Reply:
Not really convinced, but I am fine with this change.